client/rpc/rpc_sender.go

```diff
@@ -79,18 +78,16 @@ func newSender(server string, context *base.Context, retryOpts retry.Options) (*
 // through and been executed successfully. We retry here to eventually get
 // through with the same client command ID and be given the cached response.
 func (s *Sender) Send(_ context.Context, call proto.Call) {
-	retryOpts := s.retryOpts
-	retryOpts.Tag = fmt.Sprintf("rpc %s", call.Method())
-
-	if err := retry.WithBackoff(retryOpts, func() (retry.Status, error) {
+	var err error
+	for r := retry.Start(s.retryOpts); r.Next(); {
 		if !s.client.IsHealthy() {
-			return retry.Continue, nil
+			continue
 		}
 
 		method := call.Args.Method().String()
 		c := s.client.Go("Server."+method, call.Args, call.Reply, nil)
 		<-c.Done
-		if c.Error != nil {
+		if err = c.Error; err != nil {
```

Comment:
changing the `err` in the surrounding scope as a side-effect of the initializer her isn't making this easier to read. Let's just have one extra line.